code/modules/vehicles/mecha/mecha_construction_paths.dm

```diff
@@ -384,7 +384,9 @@
 	outer_plating_amount=1
 
 /datum/component/construction/mecha/gygax/action(datum/source, atom/used_atom, mob/user)
-	return INVOKE_ASYNC(src, PROC_REF(check_step), used_atom,user)
+	ASYNC //This proc will never actually sleep, it calls do_after with a time of 0.
+		. = check_step(used_atom, user)
+	return .
```

Comment:
Couldn't this just be
```
		return check_step(used_atom, user)
```

Reply:
check_step calls do_after. this triggers the linter due to SHPULD NOT SLEEP

Reply:
> Couldn't this just be

returns inside of spawn blocks is undefined behavior.

in this case i think it would actually work like a break, and move control to the end of the spawn block.

Reply:
actually...

![image](https://user-images.githubusercontent.com/7069733/218585955-fe736c50-6a93-4192-8db0-9ebcc711c532.png)

Reply:
So it's the same rules as proc arguments. Objects are pass-by-reference, everything else is copied. Interesting. Ideas?

Reply:
we shouldn't be trying to bypass the linter like this
optimally we'd have a way to mark a proc call as DOES_NOT_SLEEP, but an UNLINT could also work

The offending case here should not ever sleep, at least in the way it currently fails, but we might end up missing other cases.
S annoying.